### PR TITLE
Rename action to task

### PR DESCRIPTION
**Pull request by nhunzaker**
This commit renames Actions to Tasks. Additionally:

**Glossary document**

I've also added a glossary of terms for Microcosm. This should help us
keep things straight, and aid us in identifying areas where we can
decrease the amount of Microcosm specific stuff you need to know.

**Actions aren't going away**

Just to clarify, the idea of actions isn't gone. I just want a
stronger distinction between what you pass to repo.push() and what you
get as a result. Eventually, I want to be able to have Tasks manage
multiple actions. A Task is composed of actions, and tracks their
progress.

### CODE REVIEW

File: docs/api/glossary.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+app.
+
+**Task:** A unit of work that needs to be performed. Tasks execute the
+action they are created with. Based on the outcome of the action, they
+can move into an `open`, `update`, `resolve`, `reject`, and `cancel`
+state. As a task's state changes, a Microcosm uses Domains to
+determine how it should turn the result of the Task's work into new
+repo state. Effects can respond to changes in task states to
+perform side-effects, like saving to localStorage or tracking user
+behavior using analytics tool.
```
Comment (nhunzaker) [inline docs/api/glossary.md]: @leobauza @greypants @efatsi Feels? We could blow this out further.

**Comment by efatsi** [comment]
> Eventually, I want to be able to have Tasks manage multiple actions. A Task is composed of actions, and tracks their progress.

Can you elaborate on that?

**Comment by greypants** [comment]
Are we still calling what you `push` an action though?

**Comment by nhunzaker** [comment]
@greypants Yes. An action is what you send into `push`, no need to rename the `actions` folder :). 

@efatsi I might be ahead of myself. See https://github.com/vigetlabs/microcosm/issues/229. Basically, I want away to create a Task to be able to manage multiple sub-tasks, triggering `onDone`, `onError`, etc. based on the result of the whole.

**Comment by nhunzaker** [comment]
@efatsi For example, a confirmation warning for deleting a record. What if we could abstract that into a `confirm` action and a `deleteRecord` action?

```javascript
function confirmAndDeleteRecord (id) {
  return (task, repo) => {
    task.sequence([
      () => repo.push(confirm, 'Are you sure'), 
      () => repo.push(deleteRecord, id)
    ])
  }
}
```

I don't have ergonomics figured out here, but that's sort of where my head is at.

File: docs/api/glossary.md (hunk 1 above)
Comment (leobauza) [inline docs/api/glossary.md]: I like this, we should definitely have this. Could we move it under the `guides`/`introduction` section? Then we could break up the glossary into the different sections like `core api`, `addons`, `general stuff?`, etc

Comment (nhunzaker) [inline docs/api/glossary.md]: Cool. I like that a lot.

**Comment by efatsi** [comment]
I might be alone here, but this change confuses me. So you still `push` an `action`, but then the `action` turns into a `task`? Or a `task` becomes in charge of the `action`?

I'd like to further understand the benefit from making this nomenclature change (functionally things are staying the same right?) because it definitely brings with it the cognitive load of understanding two components instead of one for the "this is how things change" functionality.

**Comment by nhunzaker** [comment]
@efatsi I was thinking a task becomes in charge of the action. I'll hold off merging this. Let's get this right.

**Comment by nhunzaker** [comment]
Going to hold off on this for now.

**Pull request closed without merging**